### ISSUE TITLE
multiple awsnodetemplate security groups via awsIds broken between 0.30.0 and 0.31.3 

### ISSUE THREAD

**Issue by cqwense** [aws/karpenter-provider-aws#5420]
### Description

**Observed Behavior**:
In a spec with multiple awsIds for security group selector:
```Spec:
  ...
  Security Group Selector:
    Aws - Ids:  sg-0000000000000001, sg-0000000000000002, sg-0000000000000003
  ...
```

the resulting awsnodetemplate is only assigned the first security group when karpenter controller = v0.31.3 :

```
Status:
 ...
  Security Groups:
    Id:    sg-0000000000000001
    Name:  eks-cluster-sg-example-0000000001
```

**Expected Behavior**:

the awsnodetemplate should be assigned all security groups, working as expected in v0.30.0 :
```
Status:
  ...
  Security Groups:
    Id:    sg-0000000000000001
    Name:  eks-cluster-sg-example-0000000001
    Id:    sg-0000000000000002
    Name:  atmos-sg-examplesg-0000000001
    Id:    sg-0000000000000003
    Name:  atmos-sg-otherexamplesg-0000000001
```

**Reproduction Steps** (Please include YAML):

where ${awsIds} is a comma separated list of valid security group IDs, kubectl apply the following to a cluster running v0.30.0 ( may require some editing based on your environment, ex: subnet names  :

```
apiVersion: karpenter.k8s.aws/v1alpha1
kind: AWSNodeTemplate
metadata:
  name: default
spec:
  amiFamily: AL2
  amiSelector:
    aws::name: amazon-eks*-node-*
    aws::owners: amazon
  blockDeviceMappings:
    - deviceName: /dev/xvda
      ebs:
        volumeSize: 100Gi
        volumeType: gp3
        encrypted: true
  subnetSelector:
    Name: private*
  securityGroupSelector:
    aws-ids: ${awsIds}

  userData: |
    MIME-Version: 1.0
    Content-Type: multipart/mixed; boundary="BOUNDARY"

    --BOUNDARY
    Content-Type: text/x-shellscript; charset="us-ascii"

    #!/bin/sh
    set -e
    --BOUNDARY--
---

apiVersion: karpenter.sh/v1alpha5
kind: Provisioner
metadata:
  name: default
spec:
  providerRef:
    name: default
  requirements:
    - key: karpenter.k8s.aws/instance-category
      operator: In
      values: [ c ]
    - key: karpenter.k8s.aws/instance-size
      operator: In
      values: [ large, xlarge, 2xlarge ]
    - key: karpenter.sh/capacity-type
      operator: In
      values: [ on-demand ]
    - key: kubernetes.io/arch
      operator: In
      values: [ amd64 ]
    - key: kubernetes.io/os
      operator: In
      values: [ linux ]
```

confirm awsnodetemplate includes all ${awsIds} security groups

update karpenter controller to 0.31.3 

confirm awsnodetemplate includes only first ${awsIds} security group 

( if drift is enabled nodes will start to drain/replace with incorrect SGs ) 

**Versions**:
- Chart Version: 0.31.3
- Kubernetes Version (`kubectl version`): 1.25.16 / 1.24.17

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
This bug was caught and fixed here: https://github.com/aws/karpenter-provider-aws/pull/5128 and was ported into v0.32.2. I'd recommend upgrading to that version to fix this issue. Though that version contains support for beta apis, installation and behavior in that version should continue to work the same.

**Comment by jonathan-innis** [comment]
Closing this since the bug is resolved in v0.32.x
